grails-app/views/product/_mergeProductConfirmationDialog.gsp

```diff
@@ -0,0 +1,96 @@
+<div id="merge-product-confirmation-dialog" class="p-2" title="${warehouse.message(code: 'product.mergeProducts.label')}">
+    <div class="message">
+      ${warehouse.message(code: 'product.mergeProducts.confirmation.warning.label')}
+    </div>
```

Comment:
one way I though of which is a bit messy is spliting `product.mergeProducts.confirmation.warning.label` into 3 chunks 
1.  Product
2. will be merged INTO
3. _(rest of the message)_

and then render a html like:
```
${warehouse.message(code: 'chunk1)} <span id="primary-product"></span>${warehouse.message(code: 'chunk2)}<span id="duplicate-product"></span>${warehouse.message(code: 'chunk3)}
```
and later using javascript/jquery inject two of those values into spans **primary-product**, **duplicate-product**

Reply:
@drodzewicz you can pass it in model here `<g:render template="/product/mergeProductConfirmationDialog" />`

Reply:
I don't thinks so, since this `<g:render template="/product/mergeProductConfirmationDialog" />` is rendered on a server before there are any products selected

Reply:
You're right